### PR TITLE
Finished ticket #11

### PR DESCRIPTION
**Pull request by blameaimer**
-Creating test case for the new end point
-Creating end-point
-Creating and testing functions
-Considering errors
-Creating 404 test cases
-Handling the error

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -271,5 +271,72 @@ test('return an empty array because the second article has no comments ', () =>
...
+    body: "GOAT never seen a better article"
+  }
+    return request(app)
+    .post('/api/articles/1/comments')
+    .send(newComment)
+    .expect(201)
+    .then((response) => {
+      expect(response.body.comment).toEqual({
+        username: 'rogersop',
+        body: 'GOAT never seen a better article'
```
Comment (emilyb93) [inline __tests__/app.test.js:288]: I would like to see this respond with the comment_id also to prove it has been added to the database correctly

File: errors/index.js [hunk 2]
```diff
@@ -9,7 +9,11 @@ exports.handleCustomErrors = (err, req, res, next) => {
 exports.handlePSQLErrors=(err, req, res, next) => {
     if (err.code === '22P02'||err.code ==='42703'||err.code ==='23502') {
       res.status(400).send({ msg: 'Bad Request' });
-    } else next(err);
+    }
+    else if(err.code==='23503'){
+      res.status(404).send({ msg: 'Invalid article/username' });
+    }
```
Comment (emilyb93) [inline errors/index.js:15]: I would say if something is invalid, that is a bad request and should be a 400, or you need to amend your error message as 404 is reserved for valid queries that are not found

**Review by emilyb93 — CHANGES_REQUESTED** [review]
Some notes about the response here but otherwise looks good :)

**Review by maddierosecodes — APPROVED** [review]
All changes addressed :)

**Pull request merged**